### PR TITLE
Remove shebang from test files

### PR DESCRIPTION
**Pull request by budmc29**
As per the discussion in #295, this PR will remove the shebang from all the test files.

#### Todo:
- [x] lib/generator/test_template.erb
- [x] `bin/generate --all` which can be invoked with a -f option, since changes don't effect the tests
- [x] exercise/\*/\*\_test.rb removal of shebang line
- [x] `bin/executable-tests-check` should be removed
    - [x] modify `.travis.yml` so this file is not called
- [x] bin/enable-executable should be removed
- [x] bin/local-status-check needs to not reference the removed file
- [x] Run bin/setup-git-hoooks if pre-push git hook is desired, and to update with changes
- [x] Remove execution bit from all tests files
- [x] Update test files to reflect the new templates

#### How Has This Been Tested?
By running the full test suite.

#### Type of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or enhancement that would cause existing functionality to change)
- [x] Gardening (code and/or documentation cleanup)

#### Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [x] My change relies on a pending issue/pull request
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by kotp** [comment]
When this is complete, it should close #295 (and hopefully this message will accomplish that.)

**Comment by budmc29** [comment]
Might not be related, but should I remove the shebangs from this project's test/ folder as well?

**Comment by kotp** [comment]
It should probably be a different PR if you do.  That change would be related to maintaining the track, as opposed to the track itself.

File: bin/local-status-check [hunk 1]
```diff
@@ -4,8 +4,7 @@ puts "Starting #{__FILE__} checks..."
 
 commands_to_run = [
   'rake test',
```
Comment (Insti) [inline bin/local-status-check]: Unrelated to this PR, this should probably also be `bundle exec`ed

File: exercises/all-your-base/all_your_base_test.rb [hunk 2]
```diff
@@ -1,8 +1,7 @@
-#!/usr/bin/env ruby
 require 'minitest/autorun'
 require_relative 'all_your_base'
 
-# Common test data version: 1.0.0 39de23b
+# Common test data version: 1.1.0 c4d8d95
```
Comment (Insti) [inline exercises/all-your-base/all_your_base_test.rb]: The tests should not be updated other than to remove the shebang line.
You may need to checkout this file and update it manually.

Comment (budmc29) [inline exercises/all-your-base/all_your_base_test.rb]: I ran ```bin/generate --all```, should I revert that?

Comment (Insti) [inline exercises/all-your-base/all_your_base_test.rb]: It was the right thing to do for most of the generated files, just the ones with changed canonical data need reverting.

File: exercises/series/series_test.rb [hunk 3]
```diff
@@ -93,7 +92,7 @@ def test_more_complicated_slice_that_blows_up
       series.slices(slice_string.length + 1)
     end
   end
-  
+
```
Comment (Insti) [inline exercises/series/series_test.rb]: This addition is ok.

File: exercises/triangle/triangle_test.rb [hunk 4]
```diff
@@ -1,8 +1,7 @@
-#!/usr/bin/env ruby
 require 'minitest/autorun'
 require_relative 'triangle'
 
-# Common test data version: 1.0.0 fa90b35
+# Common test data version: 1.0.0 8acd78c
```
Comment (Insti) [inline exercises/triangle/triangle_test.rb]: This one is too since nothing changed in the rest of the tests.

File: exercises/all-your-base/all_your_base_test.rb (hunk 2 above)
Comment (budmc29) [inline exercises/all-your-base/all_your_base_test.rb]: Sorry, I don't think I understand.
Is there a reason to run bin/generate, but revert all the tests that were changed?

Comment (Insti) [inline exercises/all-your-base/all_your_base_test.rb]: `bin/generate --all` regenerated 44 tests, you want to keep most of those changes.
There are only a handful that need reverting.

Comment (budmc29) [inline exercises/all-your-base/all_your_base_test.rb]: Can you please list the ones I have to revert, or point me where I can find the canonical ones?

Comment (Insti) [inline exercises/all-your-base/all_your_base_test.rb]: These 5 need reverting. The updates to tests which are not just removing the shebang line should not be included in this PR.

exercises/all-your-base/all_your_base_test.rb
exercises/bowling/bowling_test.rb 
exercises/hamming/hamming_test.rb 
exercises/pangram/pangram_test.rb 
exercises/rna-transcription/rna_transcription_test.rb

**Comment by budmc29** [comment]
@Insti I have updates the files as you required.
@kotp can you please have a look and let me know if I missed anything. I consider this task completed.

**Comment by kotp** [comment]
Thank you for all the work done @budmc29.
